### PR TITLE
[EASY] Add AT_CHECK for null storage.

### PR DESCRIPTION
**Pull request by ezyang**
Signed-off-by: Edward Z. Yang <ezyang@fb.com>

### CODE REVIEW

File: aten/src/TH/THTensor.hpp [hunk 1]
```diff
@@ -136,6 +136,17 @@ inline void THTensor_setStorageOffset(THTensor* tensor, ptrdiff_t storage_offset
 
 // NB: Non-retaining
 inline THStorage* THTensor_getStoragePtr(const THTensor* tensor) {
+  // Within PyTorch, the invariant is that storage_ is always
+  // initialized; we never have tensors that don't have any storage.
+  // However, for Caffe2, this is not true, because they have permitted
+  // tensors to be allocated without specifying what scalar type
+  // they should be, only to be filled with GetMutableData is called
```
Comment (cpuhrsch) [inline aten/src/TH/THTensor.hpp]: Should it be "when GetMutableData is called"?

**Review by gchanan — APPROVED** [review]
should we guard the setters as well?

**Comment by ezyang** [comment]
Seems like a reasonable thing to do.

**Comment by weiyangfb** [comment]
@ezyang  shipit shipit

**Pull request closed without merging**